app/customer.py

```diff
@@ -0,0 +1,71 @@
+import dataclasses
+from typing import List
+
+from app.car import Car
+from app.shop import Shop
+
+
+@dataclasses.dataclass
+class Customer:
+    name: str
+    product_cart: dict
+    location: list
+    money: float
+    car: Car
+
+    @classmethod
+    def create_customers(cls, customers: List) -> List:
+        customer_objects = []
+        for customer in customers:
+            car_data = customer.pop("car")
+            car = Car(**car_data)
+            customer["car"] = car
+            customer_objects.append(cls(**customer))
+        return customer_objects
```

Comment:
U can place Car creation inside Customer init. Then create customer just using unpacking directly where you need that and remove this method
It will pretty simplify your code